fl4health/clients/mr_mtl_client.py

```diff
@@ -0,0 +1,188 @@
+        initial_global_weights = [layer_weights.detach() for layer_weights in self.init_global_model.parameters()]
+
+        assert len(initial_global_weights) == len(local_model_weights)
+        assert len(initial_global_weights) > 0
+
+        layer_inner_products: List[torch.Tensor] = [
+            torch.pow(torch.linalg.norm(initial_layer_weights - iteration_layer_weights), 2.0)
+            for initial_layer_weights, iteration_layer_weights in zip(initial_global_weights, local_model_weights)
+        ]
+
+        # network l2 inner product tensor weighted by lambda
+        return (self.lam / 2.0) * torch.stack(layer_inner_products).sum()
+
+    def get_parameters(self, config: Config) -> NDArrays:
+        """
+        For MR-MTL, we transfer the LOCAL model weights to the server to be aggregated and set as INITIAL GLOBAL model
+        weightson client side.
+
+        Args:
+            config (Config): The config is sent by the FL server to allow for customization in the function if desired.
+
+        Returns:
+            NDArrays: GLOBAL model weights to be sent to the server for aggregation
+        """
+        assert self.model is not None and self.parameter_exchanger is not None
+        return self.parameter_exchanger.push_parameters(self.model, config=config)
+
+    def set_parameters(self, parameters: NDArrays, config: Config, fitting_round: bool) -> None:
+        """
+        The parameters being pass are to be routed to the initial global model to be used in a penalty term in
+        training the local model.
+
+        Args:
+            parameters (NDArrays): Parameters have information about model state to be added to the relevant client
+                model (global model for all but the first step of MR-MTL)
+            config (Config): The config is sent by the FL server to allow for customization in the function if desired.
+            fitting_round (bool): Boolean that indicates whether the current federated learning
+                round is a fitting round or an evaluation round.
+                This is used to help determine which parameter exchange should be used for pulling parameters.
+                A full parameter exchanger is only used if the current federated learning round is the very
+                first fitting round.
+        """
+        # Make sure that the proper components exist.
+        assert self.init_global_model is not None and self.model is not None
+        assert self.parameter_exchanger is not None and isinstance(self.parameter_exchanger, FullParameterExchanger)
+
+        # Route the parameters to the GLOBAL model in MR-MTL
+        assert self.parameter_exchanger is not None
+        log(INFO, "Setting the global model weights")
+        self.parameter_exchanger.pull_parameters(parameters, self.init_global_model, config)
+
+    def update_before_train(self, current_server_round: int) -> None:
+        assert isinstance(self.model, nn.Module)
```

Comment:
Based on the fact that you're manipulating `self.init_global_model` below, I think this assert is meant to be on `self.init_global_model` instead?